sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationIdentifierSerializer.java

```diff
@@ -3,77 +3,106 @@
 
 package com.azure.android.communication.common;
 
-import java.security.InvalidParameterException;
+import android.text.TextUtils;
+
+import java.util.ArrayList;
 import java.util.Objects;
 
 class CommunicationIdentifierSerializer {
+    /**
+     * Deserialize CommunicationIdentifierModel into CommunicationIdentifier
+     * @param identifier CommunicationIdentifierModel to be deserialized
+     * @return deserialized CommunicationIdentifier
+     */
     public static CommunicationIdentifier deserialize(CommunicationIdentifierModel identifier) {
-        String id = identifier.getId();
-        CommunicationIdentifierKind kind = identifier.getKind();
+        assertSingleType(identifier);
+        String rawId = identifier.getRawId();
```

Comment:
Do not we need a `requireNonNull` here as well?

Reply:
To my understanding, rawId is only required for TeamsUser and UnknownUser. For phone number and communication user, it is optional. Am I missing anything?

Reply:
In the current schema `rawId` is required only for those two, but as a contract we want the services to always return it. So let's make it mandatory in the deserialization